### ISSUE TITLE
Not able to deploy the cluster with master data client nodes separately 

### ISSUE THREAD

**Issue by ganilmca** [opensearch-project/helm-charts#351]
Hi Team,

In the below link helm charts, we are not able to deploy the cluster with master nodes separately and data nodes.
We want dedicated master, data, and client nodes.
Can anyone tell us how to deploy the cluster like the above?

We mentioned the roles below: **in this case we could able to deploy the cluster with all roles for each pod**
roles:
  - master
  - ingest
  - data
  - remote_cluster_client

We mentioned the roles for only master nodes: **In this case, cluster was not formed**
roles:
  - master
  #- ingest
  #- data
  #- remote_cluster_client  

Thanks

**Comment by ganilmca** [comment]
Hi Team,

Is there any update on this above request?


Thanks,
Anil

**Comment by gaiksaya** [comment]
[Triage]

Hi @ganilmca,

Can you post the entire values.yaml here? 
Thanks!
cc: @prudhvigodithi @peterzhuamazon

**Comment by ganilmca** [comment]
[values.txt](https://github.com/opensearch-project/helm-charts/files/10274956/values.txt)
 Hi @gaiksaya,

Please find the attached values file, in this file we can able to deploy the cluster with all roles like(master,data & client) for every pod, but we want deploy the cluster with individual roles like dedicated master nodes & data, and client nodes.

Please help us to deploy the cluster that we want.

Thank you.

**Comment by ganilmca** [comment]
Hi Team,

Any update on this?


Thank you

**Comment by smlx** [comment]
The trick to this is to install the chart three times: once for masters, once for clients, once for data nodes. All the nodes form one Opensearch cluster despite being three helm chart installations.

**Comment by TheAlgo** [comment]
A small blog I wrote some time back where the values might be a bit different but the process will be same

Link : https://opensearch.org/blog/setup-multinode-cluster-kubernetes/

**Comment by prudhvigodithi** [comment]
Also @ganilmca just check out this [OpenSearch operator ](https://github.com/Opster/opensearch-k8s-operator), you can just have one config file to install all your components.
@idanl21 @swoehrl-mw @segalziv

**Comment by vmm-afonso** [comment]
> The trick to this is to install the chart three times: once for masters, once for clients, once for data nodes. All the nodes form one Opensearch cluster despite being three helm chart installations.

Hello there, I tried this but for whatever reason it only creates for the masters nodes and not for the data and clients. 


> A small blog I wrote some time back where the values might be a bit different but the process will be same
> 
> Link : https://opensearch.org/blog/setup-multinode-cluster-kubernetes/

I have also tried this but If I edit the roles section of my values.yml to look like this:
-------------------------------------- 
roles:
   master: "true"
   ingest: "false"
   data: "false"
   remote_cluster_client: "false"
-------------------------------------- 

I get the following error message:
-------------------------------------- 
warning: skipped value for opensearch.roles: Not a table.
Error: INSTALLATION FAILED: template: opensearch/templates/statefulset.yaml:357:21: executing "opensearch/templates/statefulset.yaml" at <has "master" .Values.roles>: error calling has: Cannot find has on type map  
-------------------------------------- 

Side Note: I also edited the nodeGroup but that didn't seem to cause any trouble 

So I just commented the roles I didn't want for my masters/data/clients and then it installed without any error messages but again, the data/client nodes were nowhere to be found

Anyone had this problem or knows how to solve this?

**Comment by vmm-afonso** [comment]
I managed to install my data and client nodes and they're running just fine I think. The master nodes however seem to be acting up.

I get the following error message:

[ERROR][o.o.b.OpenSearchUncaughtExceptionHandler] [opensearch-cluster-master-0] uncaught exception in thread [main]
org.opensearch.bootstrap.StartupException: java.lang.IllegalStateException: node does not have the data role but has shard data: [/usr/share/opensearch/data/nodes/0/indices/examplexyz]. Use 'opensearch-node repurpose' tool to clean up 

I shortened the error msg here by removing the other indices 
Where is this repurpose tool and is it possible make it a master node right from the get-go without having to use it?
